src/Schema/Values/TypeValue.php

```diff
@@ -71,7 +71,11 @@ public function cacheKey(): ?string
             // First priority: Look for a field with the @cacheKey directive
             foreach ($fieldDefinitions as $field) {
                 if (ASTHelper::hasDirective($field, CacheKeyDirective::NAME)) {
-                    return $this->cacheKey = $field->name->value;
+                    $renameDirectiveNode = ASTHelper::directiveDefinition($field, 'rename');
+
+                    return $this->cacheKey = $renameDirectiveNode
+                        ? ASTHelper::directiveArgValue($renameDirectiveNode, 'attribute')
+                        : $field->name->value;
```

Comment:
This code is duplicated and can be extracted to a function:

```suggestion
                    return $this->useFieldAsCacheKey($field);
```